### PR TITLE
add slurm test for gpu binding

### PR DESCRIPTION
**Pull request by smoors**

### CODE REVIEW

File: slurm_gpu/gpu_binding.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import pynvml
+import json
+
+
+def gpu_cpu_affinity(gpuid):
+    """
+    get GPU-CPU affinity
+    returns a list of allocated CPU ids that are on the same socket as gpuid
+    """
+    cpusetsize = math.ceil(os.cpu_count() / 64)
```
Comment (wpoely86) [inline slurm_gpu/gpu_binding.py:24]: what's this?

File: slurm_gpu/gpu_binding.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+"""
+import math
+import os
+import pynvml
+import json
+
+
+def gpu_cpu_affinity(gpuid):
+    """
+    get GPU-CPU affinity
```
Comment (wpoely86) [inline slurm_gpu/gpu_binding.py:15]: can you also document the `gpuid` paramater?

File: slurm_gpu/gpu_binding.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    aff_str = ''
+    for c_aff in pynvml.nvmlDeviceGetCpuAffinity(handle, cpusetsize):
+        aff_string = f'{c_aff:064b}{aff_str}'
+    aff_list = [int(x) for x in reversed(aff_string)]
+
+    return [i for i, aff in enumerate(aff_list) if aff != 0]
+
+
+if __name__ == '__main__':
+    pynvml.nvmlInit()
```
Comment (wpoely86) [inline slurm_gpu/gpu_binding.py:37]: please put it in a `main()` to avoid putting variables globally ;)

File: slurm_gpu/gpu_binding.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+def gpu_cpu_affinity(gpuid):
+    """
+    get GPU-CPU affinity
+    returns a list of allocated CPU ids that are on the same socket as gpuid
+    """
+    cpusetsize = math.ceil(os.cpu_count() / 64)
+    handle = pynvml.nvmlDeviceGetHandleByIndex(gpuid)
+    aff_str = ''
+    for c_aff in pynvml.nvmlDeviceGetCpuAffinity(handle, cpusetsize):
+        aff_string = f'{c_aff:064b}{aff_str}'
```
Comment (wpoely86) [inline slurm_gpu/gpu_binding.py]: you're overwrite the same var? This does need to be `aff_str = `?

File: slurm_gpu/gpu_binding.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    aff_list = [int(x) for x in reversed(aff_string)]
+
+    return [i for i, aff in enumerate(aff_list) if aff != 0]
```
Comment (wpoely86) [inline slurm_gpu/gpu_binding.py:32]: ok, I can't follow what is happening here. Can you add a couple of lines documenting this line and the ones above?

Comment (smoors) [reply slurm_gpu/gpu_binding.py:32]: i've added some comments and link to the docs, should be clear now?

File: slurm_gpu/gpu_binding.py (hunk 4 above)
Comment (smoors) [reply slurm_gpu/gpu_binding.py]: indeed, fixed

**Review by wpoely86 — APPROVED** [review]

**Pull request merged**